hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java

```diff
@@ -189,6 +190,9 @@ public ClientNetworkConfig setConnectionTimeout(int connectionTimeout) {
      * @return configured {@link com.hazelcast.client.config.ClientNetworkConfig} for chaining
      */
     public ClientNetworkConfig addAddress(String... addresses) {
+        for (String address : addresses) {
+            checkHasText(address.trim(), "member must contain text");
```

Comment:
In theory someone could pass a null as address and you will be causing an 'uncontrolled' NPE.

Reply:
Programmatically this should be very easy to accomplish, shouldn't it? Would be good to have a distinct test case for this.

Reply:
fixed and added the test.

Reply:
I think this won't help, due to the `(String... addresses)` parameter, so `config.getNetworkConfig().addAddress(null, null);` should still through an `NPE`. I think the null check has to be done in the loop.

Reply:
@Donnerbart fixed again :(